### PR TITLE
support STM32F429I-DISC1

### PR DESCRIPTION
**Pull request by TychoVrahe**
Adds support for STM32F429I-DISC1 board.

Some adjustments are done so that GUI is rectangular, and homescreen format support different dimensions. 

The resistive touch screen is quite annoying to use, and one issue that i didn't solve yet is that it sometimes does not initialize properly, and power off / on of the discovery board is needed.

### CODE REVIEW

**Comment by TychoVrahe** [comment]
should this have a changelog entry?

**Comment by matejcik** [comment]
overall style comment: don't add trailing commas to tuples (as in `if TREZOR_MODEL in ('T', 'R', )`)
the reason we write `in ('T',)` is to indicate that it's a 1-tuple (and not just the string 'T' in parens) and it's actually required in the 1 element case. not so in other cases. `(item,)` is a Python idiom, putting trailing comma elsewhere is not.

File: common/protob/messages-management.proto
Comment (mmilata) [inline common/protob/messages-management.proto:127]: Not sure about the arbitrary size screen thing. On one hand it's nice to see that the UI works reasonably well on display it was not developed for. On the other hand it would make sense to have this build as similar to model T as possible, and probably shouldn't add features to trezorlib that are not supported by anything we currently sell.

My proposal is to keep the 320x240 resolution but only support 240x240 homescreens for now, and revert the protobuf changes. Thoughts @matejcik?

File: core/SConscript.boardloader
Comment (mmilata) [inline core/SConscript.boardloader:133]: Would it still work if we merged the two branches, i.e. used STM32F427xx for disco1?

File: core/SConscript.bootloader
Comment (mmilata) [inline core/SConscript.bootloader:171]: Nit: trailing comma.

File: core/SConscript.firmware
Comment (mmilata) [inline core/SConscript.firmware:834]: Nit: trailing comma.

File: core/embed/boardloader/main.c
Comment (mmilata) [inline core/embed/boardloader/main.c:222]: Not sure I understand how is the SDRAM used, is it in addition to the on-chip RAM or instead of it? IOW, do we have more memory than on model T?

**Review by mmilata — COMMENTED** [review]
Nice work, runs fine on my board, even passed bunch of random device tests i ran on it. Couple comments inline, also please:
* add changelog entry &
* add nightly-only CI job so that we notice if it fails to build for some reason.

**Comment by mmilata** [comment]
I've added instructions for building and flashing in 742090523764cfd6e50822ec9ecd0b980efdf277, hope that's ok.

File: core/SConscript.boardloader
Comment (TychoVrahe) [reply core/SConscript.boardloader:133]: The main thing here is that Disco uses LTDC that is not available on F427. STM HAL needs to be configured accordingly. It would most likely work if we set STM32F429xx for model T and R, but it doesn't feel right

File: core/embed/boardloader/main.c
Comment (TychoVrahe) [reply core/embed/boardloader/main.c:222]: In addition to on chip RAM, this board has external RAM - 8MB more. Its used here for holding frame buffer, which wouldn't fit on chip RAM. (frame buffer is required for display with RGB interface)

File: common/protob/messages-management.proto
Comment (TychoVrahe) [reply common/protob/messages-management.proto:127]: This is not strictly needed here and we can revert the changes now, but i expect that we will run into this sooner or later- e.g. T3W1 will need different homescreen size too and we will have to differentiate somehow, and adding Jpeg320x240 seemed like worse option to me

File: core/SConscript.firmware
Comment (TychoVrahe) [reply core/SConscript.firmware]: https://github.com/trezor/trezor-firmware/commit/0b8f237c069cc390c6adaa7c36454fcb8c4e0e14

**Comment by TychoVrahe** [comment]
added changelog entries: https://github.com/trezor/trezor-firmware/commit/f19dc8a916eea72b17ab7bf61af601c406e8e9e3

though the original commit still contains `[no changelog]` as i dont want to force push right now, will try to remember to remove it

**Review by mmilata — APPROVED** [review]

File: core/embed/trezorhal/flash.h
Comment (TychoVrahe) [reply core/embed/trezorhal/flash.h]: https://github.com/trezor/trezor-firmware/commit/4b35f3b1c69c42fad8ef5fc7d5339bebccd127dd

few lines below where is `FLASH_SECTOR_STORAGE_1` defined i left it based on model, as its arbitrary selection which could, in theory at least, differ between models running on same MCU. Though the rest of the flash layout is fixed now but in theory should also be defined per model. Anyway, we will probably have to refactor this part more if we start using another MCU line where the flash is somewhat different.

File: core/embed/trezorhal/i2c.c
Comment (TychoVrahe) [reply core/embed/trezorhal/i2c.c:12]: Disco board uses I2C3, instead of I2C1 as model T. So this difference was outsourced to `trezorhal/boards/stm32f429i-disc1.h `and `trezorhal/boards/trezor_t.h`

**Review by matejcik — APPROVED** [review]

**Comment by TychoVrahe** [comment]
using D001 also in CI: https://github.com/trezor/trezor-firmware/commit/ecef0a0ee09917bab4715dd2c7d6df700c20bffe

**Comment by TychoVrahe** [comment]
rebased. fixed some issues on the way, like building core T1 variant and homescreen handling needed some adjustments after TR merge. 

will wait with merging this until vendor headers and internal_model PRs get merged too - there are some modifications with impact on this too.

**Review by hiviah — APPROVED** [review]

**Comment by TychoVrahe** [comment]
rebased.

added vendor header for DISC1, added model definition to trezorlib, in micropython this is now also recognized as DISC1 and reports as such to host. downside is that suite doesn't work with this, but i think its reasonable for this only to be supported in trezorctl for now?

also removed a condition that said which models supports passphrase on device, as in core everything now does.

**Comment by TychoVrahe** [comment]
and displaying vendor image is fixed here too.

**Comment by hiviah** [comment]
Reviewed the added commits, LGTM.

**Pull request merged**
